### DISCUSSION TITLE
Idea: Extra compiler arguments

### DISCUSSION THREAD

**Discussion by huwdp** [lightspark/lightspark#984] [Ideas]
I was thinking of adding -fsanitize=address and -D_FORTIFY_SOURCE=2 to CMake as an option that we can flip on or off. Anyone have ideas on any further compiler arguments we can use? What does everyone think?

https://developers.redhat.com/blog/2021/04/16/broadening-compiler-checks-for-buffer-overflows-in-_fortify_source
https://www.osc.edu/resources/getting_started/howto/howto_use_address_sanitizer

**Comment by dbluelle** [comment]
I'm for it, everything that helps in finding and fixing bugs is appreciated!
Currently most crashes/memleaks come from issues with reference counting of the actionscript objects, so I don't know if that will help in those cases...
I think these flags should be set to false by default, as they will most likely cause a loss in performance.
